### ISSUE TITLE
Integrate with pre-commit tool

### ISSUE THREAD

**Issue by adithyabsk** [troessner/reek#1527]
## Feature

It would be really nice if the project integrated with the [pre-commit project](https://pre-commit.com/), a widely used pre-commit hook management tool. Only a single file is required for integration (the `.pre-commit-hooks.yaml` file)

## Solution

I've already set up a project mirror doing exactly that would suggest a link to my mirror (which auto-sync to the mainline project's releases)

https://github.com/adithyabsk/mirrors-reek

Based on #457 it seems that a separate project is preferred in this case?

## Next steps
**Option A (most preferred)**
Move forward with integrating pre-commit tool using my project's mirror as a starting point
**Option B**
Add a link in the docs to suggest using the mirror one is interested in using pre-commit hooks
**Option C (least preferred)**
Do nothing 😦

**Comment by mvz** [comment]
First off, I think using reek in a pre-commit hook is a very bad idea! I'm now wondering whether we should even point people to overcommit.

Why is it a bad idea? I think you should generally only check things that are trivial to fix without accidentally breaking your code. If you need to refactor to fix things, don't check it in a pre-commit hook. Committing after every step is a vital part of a safe refactoring process.

Adding support for this tool in the Reek repo would add a maintenance burden to Reek's developers, since we would need to understand this tool and its configuration system. We would also need to know how to check if we haven't broken this configuration, and how to update it if pre-commit itself changes.

Since pre-commit's maintainers and/or users know all these things, it makes more sense to maintain support for Reek in pre-commit outside of the Reek repository.

In conclusion, my preference would got to either option B or option C.

PS: That's a very odd use of the term 'mirror'!

**Comment by adithyabsk** [comment]
> using reek in a pre-commit hook is a very bad idea

@mvz thanks for the thoughtful response. One thing to note is that although the project's name is pre-commit, the tool could be configured to run at different stages (ie pre-push). 

> add a maintenance burden to Reek's developers

Totally, understand the burden aspect to the developers, though there isn't much to it besides adding a single configuration file that pre-commit: `.pre-commit-hooks.yaml`. As far as the mirror naming goes, it's just what the pre-commit project convention seems to be as it is "mirroring" being able to install the tool directly from the project. Totally _not_ a mirror of the project!

> makes more sense to maintain support for Reek in pre-commit outside of the Reek repository.

Either way, if the project isn't interested in supporting pre-commit, that's totally understandable. Would you be opposed to a pull request pointing people interested in commit-hooks in the direction of the configuration project that I have already put together?

**Comment by mvz** [comment]
> Would you be opposed to a pull request pointing people interested in commit-hooks in the direction of the configuration project that I have already put together?

Such a pull request would be totally fine.

**Comment by adithyabsk** [comment]
@mvz cool, just created a PR #1530

**Comment by mvz** [comment]
Closing this since you retracted your pull request.
